IPv6 with Redis

I have Redis and Sentinels running on IPv6.

When configuring redishappy-consul, I set "[::1]" for the Sentinel and Consul addresses, which seems to work.

After that though, I start getting this error:

```
ERROR: 2016/06/15 08:06:05 manager.go:83: Error starting monitor fdf1:a315:9433:8014:f816:3eff:fe1a:288c:26379 : dial tcp: too many colons in address fdf1:a315:9433:8014:f816:3eff:fe1a:288c:26379
ERROR: 2016/06/15 08:06:05 pubsubclient.go:18: PubSubClient Error connecting to fdf1:a315:9433:8014:f816:3eff:fef9:96c3:26379 : dial tcp: too many colons in address fdf1:a315:9433:8014:f816:3eff:fef9:96c3:26379
```

--- a/configuration/configuration.go
+++ b/configuration/configuration.go
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"errors"
 	"io/ioutil"
+	"net"
 	"os"
 	"strconv"
 	"strings"
@@ -143,12 +144,19 @@ func overrideClusterConfiguration(config *Configuration) {
 
 		for _, clusterConfig := range bits {
 
-			ok, host, port := fauxHostAndIpToBits(clusterConfig)
+			host, port, err := net.SplitHostPort(clusterConfig)
 
-			if !ok {
+			if err != nil {
 				logger.Error.Panicf("Error parsing port REDISHAPPY_CLUSTERS : %s {%s}", env, clusterConfig)
 			}
-			config.Clusters = append(config.Clusters, types.Cluster{Name: host, ExternalPort: port})
+
+			portAsInt, err := strconv.Atoi(port)
+
+			if err != nil {
+				logger.Error.Panicf("Error parsing port REDISHAPPY_CLUSTERS : %s {%s}", env, clusterConfig)
+			}
+
+			config.Clusters = append(config.Clusters, types.Cluster{Name: host, ExternalPort: portAsInt})
 		}
 		logger.Info.Printf("Using environment override for cluster configuration REDISHAPPY_CLUSTERS : %s", env)
 	}
@@ -165,30 +173,19 @@ func overrideSentinelConfiguration(config *Configuration) {
 
 		for _, sentinelConfig := range bits {
 
-			ok, host, port := fauxHostAndIpToBits(sentinelConfig)
+			host, port, err := net.SplitHostPort(sentinelConfig)
 
-			if !ok {
+			if err != nil {
 				logger.Error.Panicf("Error parsing port REDISHAPPY_SENTINELS : %s {%s}", env, sentinelConfig)
 			}
-			config.Sentinels = append(config.Sentinels, types.Sentinel{Host: host, Port: port})
+			portAsInt, err := strconv.Atoi(port)
+
+			if err != nil {
+				logger.Error.Panicf("Error parsing port REDISHAPPY_SENTINELS : %s {%s}", env, sentinelConfig)
+			}
+			config.Sentinels = append(config.Sentinels, types.Sentinel{Host: host, Port: portAsInt})
 
 		}
 		logger.Info.Printf("Using environment override for sentinel configuration REDISHAPPY_SENTINELS : %s", env)
 	}
-}
-
-func fauxHostAndIpToBits(hostAndIp string) (bool, string, int) {
-	bits := strings.Split(hostAndIp, ":")
-
-	if len(bits) != 2 {
-		return false, "", 0
-	}
-
-	port, err := strconv.Atoi(bits[1])
-
-	if err != nil {
-		return false, "", 0
-	}
-
-	return true, bits[0], port
 }

--- a/main/redis-consul/consul_flipperclient.go
+++ b/main/redis-consul/consul_flipperclient.go
@@ -3,7 +3,6 @@ package main
 import (
 	"fmt"
 
-	"github.com/armon/consul-api"
 	"github.com/mdevilliers/redishappy/configuration"
 	"github.com/mdevilliers/redishappy/services/logger"
 	"github.com/mdevilliers/redishappy/types"

--- a/main/redis-consul/consul_flipperclient.go
+++ b/main/redis-consul/consul_flipperclient.go
@@ -3,6 +3,7 @@ package main
 import (
 	"fmt"
 
+	consulapi "github.com/armon/consul-api"
 	"github.com/mdevilliers/redishappy/configuration"
 	"github.com/mdevilliers/redishappy/services/logger"
 	"github.com/mdevilliers/redishappy/types"

--- a/types/sentinel.go
+++ b/types/sentinel.go
@@ -1,7 +1,8 @@
 package types
 
 import (
-	"fmt"
+	"net"
+	"strconv"
 )
 
 type Sentinel struct {
@@ -10,5 +11,5 @@ type Sentinel struct {
 }
 
 func (s *Sentinel) GetLocation() string {
-	return fmt.Sprintf("%s:%d", s.Host, s.Port)
+	return net.JoinHostPort(s.Host, strconv.Itoa(s.Port))
 }

